nix/src/main/java/org/gnode/nix/File.java

```diff
@@ -0,0 +1,201 @@
+package org.gnode.nix;
+
+import org.bytedeco.javacpp.IntPointer;
+import org.bytedeco.javacpp.Loader;
+import org.bytedeco.javacpp.Pointer;
+import org.bytedeco.javacpp.annotation.*;
+
+import java.util.ArrayList;
+
+@Platform(value = "linux",
+        include = {"<nix/File.hpp>"},
+        link = {"nix"},
+        linkpath = {"/usr/local/lib/"},
+        includepath = {"/usr/local/include/"})
+@Namespace("nix")
+public class File extends Pointer {
+
+    static {
+        Loader.load();
+    }
+
+    //--------------------------------------------------
+    // Constructors
+    //--------------------------------------------------
+
+    /**
+     * Constructor that creates an uninitialized File.
+     * <p/>
+     * Calling any method on an uninitialized file will throw a {@link java.lang.RuntimeException}.
+     */
+    public File() {
+        allocate();
+    }
+
+    private native void allocate();
+
+    /**
+     * Copy constructor.
+     *
+     * @param other The file to copy.
+     */
+    public File(@Const @ByRef File other) {
```

Comment:
Due to reference counting in the NIX library the copy ctor might be harmful. If it is not strictly necessary omit the ctor.